flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -646,6 +646,10 @@ public String toString() {
                 }
 
                 final InputChannel inputChannel = inputChannelOpt.get();
+                if (inputChannel.isReleased()) {
```

Comment:
At this point it's probably more plausible to just return empty on `inputChannel.getNextBuffer()` to avoid reading the `AtomicBoolean` twice.